### PR TITLE
Processors: don't default padding side

### LINKED ISSUE #33847: Llava processor changing padding side of tokenizer has no effect.
### System Info

When manually setting `processor.tokenizer.padding_side` to a different value than its init value for `LlavaProcessor` (also happens with `LlavaNextProcessor`, `LlavaOnevisionProcessor` and most likely all processor from Llava family), the padding side when using `processor(texts=...)` remains unchanged.

This is due to the following being used in the `__call__` method:
```python
output_kwargs = self._merge_kwargs(
            LlavaProcessorKwargs,
            tokenizer_init_kwargs=self.tokenizer.init_kwargs,
            **kwargs,
        )
[...]
text_inputs = self.tokenizer(prompt_strings, **output_kwargs["text_kwargs"])
```
Since the original padding side is contained in `self.tokenizer.init_kwargs`, it will then be passed to the tokenizer call through `**output_kwargs["text_kwargs"]`. One can go around the issue by forcing calling `processor(texts=texts, padding_side="right")` but I think it would be better that the processor behaves as expected after `processor.tokenizer.padding_side` is changed from its init value.

- `transformers` version: 4.45.0
- Platform: Linux-5.15.0-78-generic-x86_64-with-glibc2.35
- Python version: 3.11.9
- Huggingface_hub version: 0.25.1
- Safetensors version: 0.4.5
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.4.0+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: No
- Using GPU in script?: No
- GPU type: NVIDIA H100 80GB HBM3

### Who can help?

@zucchini-nlp @LysandreJik 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from transformers import AutoProcessor

processor = AutoProcessor.from_pretrained("llava-hf/llava-v1.6-mistral-7b-hf")
texts = ["hi there", "hi there longer"]
outputs = processor(text=texts, padding=True)
print(outputs.attention_mask)
processor.tokenizer.padding_side = "right"
outputs = processor(text=texts, padding=True)
print(outputs.attention_mask)
```

### Expected behavior

With the current code we get:
[[0, 1, 1, 1], [1, 1, 1, 1]]
[[0, 1, 1, 1], [1, 1, 1, 1]]

(No changes even after changing padding side of tokenizer).

Should print the following:
[[0, 1, 1, 1], [1, 1, 1, 1]]
[[1, 1, 1, 0], [1, 1, 1, 1]]

After changing padding side to right, we expect to see the 0 corresponding to pad token in attention mask to the right.

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/33847. We should not have default values for padding side after the last PR that enabled passing `padding_side` to tokenizer directly. Users are free to set padding side as tokenizer attr or pass it to processors directly

cc @yonigozlan if you had any PRs in progress with default padding-side 😉

### CODE REVIEW

File: src/transformers/models/llava/processing_llava.py [hunk 1]
```diff
@@ -30,9 +30,7 @@
 
 class LlavaProcessorKwargs(ProcessingKwargs, total=False):
     _defaults = {
-        "text_kwargs": {
-            "padding": False,
-        },
```
Comment (molbap) [inline src/transformers/models/llava/processing_llava.py:35]: This was the default kwarg value initially in the processor, wasn't it? I might be mistaken but I thought it was padding_side that was culprit

**Review by molbap — COMMENTED** [review]
Will be cleaner! added a single comment

File: src/transformers/models/llava/processing_llava.py (hunk 1 above)
Comment (zucchini-nlp) [reply src/transformers/models/llava/processing_llava.py:35]: my bad, saw `padding_side` here somehow hehe, will fix back

**Review by ArthurZucker — APPROVED** [review]
➕ on Pablo's comment, good to go otherwise!

File: src/transformers/processing_utils.py [hunk 2]
```diff
@@ -829,7 +829,12 @@ class MyProcessingKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwarg
             for modality_key in ModelProcessorKwargs.__annotations__[modality].__annotations__.keys():
                 # init with tokenizer init kwargs if necessary
                 if modality_key in tokenizer_init_kwargs:
-                    default_kwargs[modality][modality_key] = tokenizer_init_kwargs[modality_key]
+                    value = (
+                        getattr(self.tokenizer, modality_key)
+                        if hasattr(self.tokenizer, modality_key)
+                        else tokenizer_init_kwargs[modality_key]
+                    )
+                    default_kwargs[modality][modality_key] = value
```
Comment (zucchini-nlp) [inline src/transformers/processing_utils.py:837]: I just realized the issue comes from this line when we use default tokenizer kwargs that were set during init. It a bit breaking if ppl change the attr like `padding_side` and that is not reflected in init_kwargs, so I think we can check first tokenizer attr and then fallback to the init kwargs
@molbap wdyt?

Comment (molbap) [reply src/transformers/processing_utils.py:837]: Ah I see, yes that makes sense, it should be a "soft" check though to make sure we are not requiring anything at this stage, just that if it is present, it should be superseding. 
Handling priorities in this script is :zany_face: good catch though!

Comment (zucchini-nlp) [reply src/transformers/processing_utils.py:837]: yea, i think the check is soft as we first see if attribute if present, and if not fallback to the usual `init_kwargs[key]`
